### DISCUSSION TITLE
how to use (user table) in serenity for login in homepage

### DISCUSSION THREAD

**Discussion by samaremad** [serenity-is/Serenity#5374] [Help]
hello everyone 

 i already created and generated (user table) in serenity  and now i want to use this information in login page (for general user not for admin panel users)

how to do that?

![Screenshot (38)](https://user-images.githubusercontent.com/77009503/105321853-b13fb680-5bd0-11eb-8981-82d705715f3b.png)

![Screenshot (39)](https://user-images.githubusercontent.com/77009503/105321877-bc92e200-5bd0-11eb-8685-d0307177452d.png)

**Comment by minhhungit** [comment]
@samaremad  If you just want to modify login code, you can modify the code here

```csharp
\Modules\Administration\User\Authentication\AuthenticationService.cs
```

You can even create yourself behavior by implement `IAuthenticationService` and registry it like this

```csharp
services.AddSingleton<IAuthenticationService, Administration.AuthenticationService>();
```

btw, I don't know your business logic but I think using a difference user table like that is not necessary and can not reuse serenity buit-in features like check permission..., if I do I will add more columns to exist user table

I also see password column on new table is not encrypted, you should encrypt it :)
